appOPHD/UI/FileIo.cpp

```diff
@@ -180,12 +180,21 @@ void FileIo::onClose()
 
 void FileIo::onFileIo()
 {
-	mSignal(mFileName.text(), mMode);
+	if(mMode == FileOperation::Save)
+	{
+		mSaveSignal(mFileName.text());
+	}
+
+	if(mMode == FileOperation::Load)
+	{
+		mSignal(mFileName.text(), mMode);
+	}
```

Comment:
Minor point: This technically breaks some old uses of `mSignal`, since existing client code will still expect save events to be delivered to the old signal. It's sometimes nice to provide a transition point where the new feature becomes functional, without breaking the old feature. That can be done by delivering the event to both `Signal` objects. Client code would only ever really be designed to listen to one of them. By keeping both functional, there's an easy transition period where client code can be updated at their own pace, without being forced to update. Though considering the context of the whole PR, I expect the old signal to be removed by the end, so we're talking about a really short transition period here. It's almost not worth mentioning.